src/app-layer-htp.c

```diff
@@ -824,8 +822,7 @@ static int HTPHandleResponseData(Flow *f, void *htp_state,
 
     htp_time_t ts = { f->lastts.tv_sec, f->lastts.tv_usec };
     if (input_len > 0) {
-        r = htp_connp_res_data(hstate->connp, &ts, input, input_len);
-        switch(r) {
+        switch(htp_connp_res_data(hstate->connp, &ts, input, input_len)) {
```

Comment:
I don't like this, the compiler can optimize this and while debugging it's good to have an explicit variable to inspect.